ddtrace/contrib/botocore/patch.py

```diff
@@ -139,9 +288,17 @@ def patched_api_call(original_func, instance, args, kwargs):
                 if endpoint_name == "lambda" and operation == "Invoke":
                     inject_trace_to_client_context(args, span)
                 if endpoint_name == "sqs" and operation == "SendMessage":
-                    inject_trace_to_sqs_message(args, span)
+                    inject_trace_to_sqs_or_sns_message(args, span)
                 if endpoint_name == "sqs" and operation == "SendMessageBatch":
-                    inject_trace_to_sqs_batch_message(args, span)
+                    inject_trace_to_sqs_or_sns_batch_message(args, span)
+                if endpoint_name == "events" and operation == "PutEvents":
```

Comment:
I wonder if we should guard the context injection by AWS service with a config option (`botocore_sqs_distributed_tracing`, `botocore_kinesis_distributed_tracing`, etc).

The concern is that if there is a need to not propagate context for one or several of the AWS services but not all (performance issues, etc).

We could still have the general `distributed_tracing` flag but then also provide ones to disable specific ones.

Reply:
This is a great idea, I can push this out in a separate PR after this release. This was discussed internally within our team as well and we decided to do this down the line once we got some feedback from customers. Doing this now would also require doing this for the JS tracer which is not possible with the current timeline for the Managed Service/Inferred Spans project.